.circleci/config.yml

```diff
@@ -582,6 +582,63 @@ jobs:
           working_directory: /pika/build
           when: always
 
+  arm64_build:
+    machine:
+      image: ubuntu-2004:current
+    resource_class: arm.large
+    steps:
+      - checkout
+      - run:
+          name: Building Arm64
+          command: |
+            # Check architecture = arm
+            uname -m
+            # Install CMake
+            wget -q https://cmake.org/files/v3.22/cmake-3.22.6-linux-aarch64.tar.gz \
+              -O cmake.tar.gz && echo \
+              "79be85d3e76565faacd60695cee11d030f7e7dd8691476144fa25eb93dbd0397  cmake.tar.gz" \
+              | sha256sum --check --status && \
+              sudo tar --strip-components=1 -C /usr/local -xf cmake.tar.gz && \
+              rm cmake.tar.gz
+            mkdir build
+            git clone https://github.com/pika-org/pika.git
+            cmake --version
+            sudo apt install clang ninja-build libboost-regex-dev libboost-context-dev libhwloc-dev mpi-default-dev -y
+            cd build
+            cmake \
+                /home/circleci/project/pika \
+                -G "Ninja" \
+                -DCMAKE_BUILD_TYPE=Debug \
+                -DCMAKE_CXX_COMPILER=clang++ \
+                -DCMAKE_CXX_FLAGS="-fcolor-diagnostics" \
+                -DPIKA_WITH_GIT_COMMIT=${CIRCLE_SHA1} \
+                -DPIKA_WITH_GIT_BRANCH="${CIRCLE_BRANCH}" \
+                -DPIKA_WITH_GIT_TAG="${CIRCLE_TAG}" \
+                -DPIKA_WITH_CXX_STANDARD=17 \
+                -DPIKA_WITH_PRECOMPILED_HEADERS=On \
+                -DPIKA_WITH_TOOLS=On \
+                -DPIKA_WITH_MALLOC=system \
+                -DPIKA_WITH_MPI=ON \
+                -DPIKA_WITH_EXAMPLES=On \
+                -DPIKA_WITH_GENERIC_CONTEXT_COROUTINES=On \
+                -DPIKA_WITH_TESTS=On \
+                -DPIKA_WITH_TESTS_HEADERS=On \
+                -DPIKA_WITH_PARALLEL_TESTS_BIND_NONE=On \
+                -DPIKA_WITH_COMPILER_WARNINGS=On \
+                -DPIKA_WITH_COMPILER_WARNINGS_AS_ERRORS=On \
+                -DPIKA_WITH_DEPRECATION_WARNINGS=On \
+                -DPIKA_WITH_THREAD_LOCAL_STORAGE=On \
+                -DPIKA_WITH_STACKTRACES_STATIC_SYMBOLS=On \
+                -DPIKA_WITH_STACKTRACES_DEMANGLE_SYMBOLS=Off \
+                -DPIKA_WITH_TESTS_DEBUG_LOG=On \
+                -DPIKA_WITH_TESTS_DEBUG_LOG_DESTINATION=/build/debug-log-arm.txt \
+                -DPIKA_WITH_SPINLOCK_DEADLOCK_DETECTION=On
+            ninja -j4 -k 0 pika
```

Comment:
Could you please split this up into "Install dependencies", "Configure", "Build", and a new "Test" step? The "Test" step is optional to get working for this PR. I'd just like to see how good/bad the test runs actually are. If it works right away let's keep it in this PR. If there's _anything_ that looks off we delay it for a separate PR.

TL;DR. Splitting into multiple steps: required for this PR. Building and running tests: optional.